bindings/sundials4py/include/sundials4py_helpers.hpp

```diff
@@ -0,0 +1,164 @@
+/*------------------------------------------------------------------------------
+ * Programmer(s): Cody J. Balos @ LLNL
+ *------------------------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2025-2025, Lawrence Livermore National Security,
+ * University of Maryland Baltimore County, and the SUNDIALS contributors.
+ * Copyright (c) 2013-2025, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * Copyright (c) 2002-2013, Lawrence Livermore National Security.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ *----------------------------------------------------------------------------*/
+
+#ifndef _SUNDIALS4PY_HELPERS_HPP
+#define _SUNDIALS4PY_HELPERS_HPP
+
+#include "sundials4py.hpp"
+
+namespace nb = nanobind;
+
+namespace sundials4py {
+
+/// This function will call a user-supplied Python function through C++ side wrappers
+/// \tparam FnType is the function signature, e.g., std::remove_pointer_t<CVRhsFn>
+/// \tparam FnTableType is the struct function table that holds the user-supplied Python functions as std::function
+/// \tparam UserDataArg is the index of the void* user_data argument of the C function. We are counting from the last arg to the first arg, so if user_data is the last arg then this should be 1.
```

Comment:
Wrap long lines (surprised clang format didn't catch this in the `check-litgen` workflow)